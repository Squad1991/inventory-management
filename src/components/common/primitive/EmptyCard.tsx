import { css } from '@compiled/react';
import React from 'react';
import Stack from '~src/components/common/primitive/Stack';
import { ComponentPropsWithTestId } from '~src/components/common/utils/types';

interface EmptyCardProps extends ComponentPropsWithTestId {
  message: React.ReactNode;
  width?: string;
  height?: string;
}

const EmptyCardStyles = css({
  justifyContent: 'center',
  flex: '1',
  alignItems: 'center',
});

const EmptyCard: React.FC<EmptyCardProps> = ({ message, height, width, testId }) => {
  return (
    <Stack data-testid={testId} css={[EmptyCardStyles, css({ height, width })]}>
      {message}
    </Stack>
  );
};

export default EmptyCard;
