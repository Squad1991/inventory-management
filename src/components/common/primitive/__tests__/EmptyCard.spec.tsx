import { render } from '@testing-library/react';
import React from 'react';
import EmptyCard from '../EmptyCard';

describe('EmptyCard', () => {
  it('renders correctly with text msg', () => {
    const tree = render(<EmptyCard message={'empty card'} />);
    expect(tree).toMatchSnapshot();
  });
});
