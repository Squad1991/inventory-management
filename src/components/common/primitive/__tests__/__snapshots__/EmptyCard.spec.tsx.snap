// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyCard > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-cmpld="true"
      >
        ._1bah1h6o{justify-content:center}._16jlkb7n{flex-grow:1}._1o9zkb7n{flex-shrink:1}._i0dlf1ug{flex-basis:0%}._4cvr1h6o{align-items:center}._4t3imry4{height:var(--_roksyz)}._1bsb9tg7{width:var(--_1ea5ebz)}
      </style>
      <style
        data-cmpld="true"
      >
        ._1e0c1txw{display:flex}._2lx21bp4{flex-direction:column}
      </style>
      <div
        class="_1e0c1txw _2lx21bp4 _1bah1h6o _16jlkb7n _1o9zkb7n _i0dlf1ug _4cvr1h6o _4t3imry4 _1bsb9tg7"
        style="--_roksyz: var(--c-, ); --_1ea5ebz: var(--c-, );"
      >
        empty card
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-cmpld="true"
    >
      ._1bah1h6o{justify-content:center}._16jlkb7n{flex-grow:1}._1o9zkb7n{flex-shrink:1}._i0dlf1ug{flex-basis:0%}._4cvr1h6o{align-items:center}._4t3imry4{height:var(--_roksyz)}._1bsb9tg7{width:var(--_1ea5ebz)}
    </style>
    <style
      data-cmpld="true"
    >
      ._1e0c1txw{display:flex}._2lx21bp4{flex-direction:column}
    </style>
    <div
      class="_1e0c1txw _2lx21bp4 _1bah1h6o _16jlkb7n _1o9zkb7n _i0dlf1ug _4cvr1h6o _4t3imry4 _1bsb9tg7"
      style="--_roksyz: var(--c-, ); --_1ea5ebz: var(--c-, );"
    >
      empty card
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`EmptyCard > renders correctly with text msg 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-cmpld="true"
      >
        ._1bah1h6o{justify-content:center}._16jlkb7n{flex-grow:1}._1o9zkb7n{flex-shrink:1}._i0dlf1ug{flex-basis:0%}._4cvr1h6o{align-items:center}._4t3imry4{height:var(--_roksyz)}._1bsb9tg7{width:var(--_1ea5ebz)}
      </style>
      <style
        data-cmpld="true"
      >
        ._1e0c1txw{display:flex}._2lx21bp4{flex-direction:column}
      </style>
      <div
        class="_1e0c1txw _2lx21bp4 _1bah1h6o _16jlkb7n _1o9zkb7n _i0dlf1ug _4cvr1h6o _4t3imry4 _1bsb9tg7"
        style="--_roksyz: var(--c-, ); --_1ea5ebz: var(--c-, );"
      >
        empty card
      </div>
    </div>
  </body>,
  "container": <div>
    <style
      data-cmpld="true"
    >
      ._1bah1h6o{justify-content:center}._16jlkb7n{flex-grow:1}._1o9zkb7n{flex-shrink:1}._i0dlf1ug{flex-basis:0%}._4cvr1h6o{align-items:center}._4t3imry4{height:var(--_roksyz)}._1bsb9tg7{width:var(--_1ea5ebz)}
    </style>
    <style
      data-cmpld="true"
    >
      ._1e0c1txw{display:flex}._2lx21bp4{flex-direction:column}
    </style>
    <div
      class="_1e0c1txw _2lx21bp4 _1bah1h6o _16jlkb7n _1o9zkb7n _i0dlf1ug _4cvr1h6o _4t3imry4 _1bsb9tg7"
      style="--_roksyz: var(--c-, ); --_1ea5ebz: var(--c-, );"
    >
      empty card
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
